test/IRGen/copy_to_ref.sil

```diff
@@ -0,0 +1,147 @@
+// RUN: %swift %s -emit-ir -O -module-name run | %FileCheck %s
+
+sil_stage canonical
+
+import Builtin
+import Swift
+
+class SingleInt {
+  var x : Int
+}
+
+// CHECK-LABEL: define swiftcc void @test_int
```

Comment:
As you can see, even without any added optimizations, the llvm-ir codegen is substantially better.